frontend/src/reducers/ModuleEditorContextReducer.ts

```diff
@@ -15,7 +15,8 @@ const createLesson = (
   state: EditorStateType,
   lesson: LessonType,
 ): EditorStateType => {
-  const newState = { ...state };
+  // Create deep copy of state
+  const newState = JSON.parse(JSON.stringify(state));
```

Comment:
It's because it seems state has readonly properties, specifically state.course.modules[moduleIndex].lessons was readonly and using a deep copy circumvented that.

Reply:
Yeah not a fan of this solution but it's fine for now. We will put it in the todos.